pkg/webui/console/components/gateway-visibility-form/index.js

```diff
@@ -0,0 +1,134 @@
+
+import gatewayVisibilityMessages from '@console/lib/packet-broker/messages'
+
+const m = defineMessages({
+  saveDefaultGatewayVisibility: 'Save default gateway visibility',
+})
+
+const validationSchema = Yup.object({
+  visibility: Yup.object({}),
+})
+
+const GatewayVisibilityForm = ({ onSubmit, initialValues, error, submitMessage }) => {
+  const handleSubmit = useCallback(values => onSubmit(validationSchema.cast(values)), [onSubmit])
+
+  return (
+    <Form
+      onSubmit={handleSubmit}
+      initialValues={initialValues}
+      error={error}
+      validationSchema={validationSchema}
+    >
+      <Row>
+        <Col md={6} xs={12}>
+          <Form.Field
+            name="visibility.location"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayLocationLabel}
+            description={gatewayVisibilityMessages.gatewayLocationDescription}
+          />
+          <Form.Field
+            name="visibility.antenna_placement"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayAntennaPlacementLabel}
+            description={gatewayVisibilityMessages.gatewayAntennaPlacementDescription}
+          />
+          <Form.Field
+            name="visibility.antenna_count"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayAntennaCountLabel}
+            description={gatewayVisibilityMessages.gatewayAntennaCountDescription}
+          />
+          <Form.Field
+            name="visibility.fine_timestamps"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayFineTimestampsLabel}
+            description={gatewayVisibilityMessages.gatewayFineTimestampsDescription}
+          />
+        </Col>
+        <Col sm={6} xs={12}>
+          <Form.Field
+            name="visibility.contact_info"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayContactInfoLabel}
+            description={gatewayVisibilityMessages.gatewayContactInfoDescription}
+          />
+          <Form.Field
+            name="visibility.status"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayStatusLabel}
+            description={gatewayVisibilityMessages.gatewayStatusDescription}
+          />
+          <Form.Field
+            name="visibility.frequency_plan"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayFreqPlanLabel}
+            description={gatewayVisibilityMessages.gatewayFreqPlanDescription}
+          />
+          <Form.Field
+            name="visibility.packet_rates"
+            component={Checkbox}
+            label={gatewayVisibilityMessages.gatewayPacketRatesLabel}
+            description={gatewayVisibilityMessages.gatewayPacketRatesDescription}
+          />
+        </Col>
+      </Row>
+      <SubmitBar>
+        <Form.Submit component={SubmitButton} message={submitMessage} />
```

Comment:
```suggestion
        <Form.Submit component={SubmitButton} message={m.saveDefaultGatewayVisibility} />
```
Not required to pass the message as a prop, unless you actually need to be able to control the button message for different contexts.